File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int cnt = 0;
    
    public int solution(int[] numbers, int target) {
        int answer = 0;
        
        dfs(numbers, target, 1, numbers[0]);
        dfs(numbers, target, 1, -1 * numbers[0]);

        answer = cnt;
        
        return answer;
    }
    
    public static void dfs(int[] numbers, int target, int level, int sum) {
        if(level == numbers.length) {
            if(sum == target) {
                cnt++;
            }
            return;
        }
        
        dfs(numbers, target, level + 1, sum + numbers[level]);
        dfs(numbers, target, level + 1, sum - numbers[level]);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
        
    const dfs = (lev = 0, num = 0) => {
        if(lev >= numbers.length) {
            if(num == target) {
                answer++
            }
            return
        }
        
        dfs(lev+1, num + numbers[lev])
        dfs(lev+1, num - numbers[lev])
    }
    
    dfs()
    
    return answer;
}